lib/workers/global/config/parse/index.ts

```diff
@@ -22,6 +22,27 @@ export async function parseConfigs(
   const cliConfig = cliParser.getConfig(argv);
   const envConfig = envParser.getConfig(env);
 
+  if (cliConfig?.dryRun === 'true') {
+    logger.warn('cli config dryRun property has been changed to full');
+    cliConfig.dryRun = 'full';
+  }
+  if (envConfig?.dryRun === 'true') {
+    logger.warn('env config dryRun property has been changed to full');
+    envConfig.dryRun = 'full';
+  }
+  if (cliConfig?.dryRun === 'false' || cliConfig?.dryRun === 'null') {
+    logger.warn(
+      'cli config dryRun property has been changed to null, running with normal mode.'
+    );
+    cliConfig.dryRun = null;
+  }
+  if (envConfig?.dryRun === 'false' || envConfig?.dryRun === 'null') {
+    logger.warn(
+      'env config dryRun property has been changed to null, running with normal mode.'
+    );
+    envConfig.dryRun = null;
+  }
```

Comment:
Why do warn when set to null? I usally have `dryRun=true` in my `config.js` and override on cli with `--dry-run false` when developping / debugging locally

```suggestion
  if (cliConfig?.dryRun === 'false') {
    logger.warn(
      'cli config dryRun property has been changed to null, running with normal mode.'
    );
    cliConfig.dryRun = null;
  }
  if (envConfig?.dryRun === 'false') {
    logger.warn(
      'env config dryRun property has been changed to null, running with normal mode.'
    );
    envConfig.dryRun = null;
  }
```

@rarkins Do we really like to warn here? I prefer to simply migrate, as we do with other options.

Reply:
I've done some testing and those conditions can be removed, as it seems the options are migrated before.
![image](https://user-images.githubusercontent.com/1798109/161694318-9230a61f-98fe-4e78-ad73-1bac689b6d60.png)

Reply:
@viceice in case we want to remove these conditions, where should I add migration for these cases ? is it related to the migration I did in config.js ?

Reply:
It would be nice to warn so that one day we can remove all the migrations. Ie warning serves as deprecating

Reply:
should I remove these conditions ? and to add the cli migrations to :
https://github.com/renovatebot/renovate/blob/2120171e6729f200ae0cff191c17ea92792ffad7/lib/workers/global/config/parse/cli.ts#L16-L31

Reply:
Go ahead

Reply:
@rarkins I pushed the changes